envoy/network/transport_socket.h

```diff
@@ -310,6 +318,21 @@ class UpstreamTransportSocketFactory : public virtual TransportSocketFactoryBase
    */
   virtual void hashKey(std::vector<uint8_t>& key,
                        TransportSocketOptionsConstSharedPtr options) const PURE;
+
+  /*
+   * @return the pointer to the SSL context, or nullptr for non-TLS factories.
+   */
+  virtual Envoy::Ssl::ClientContextSharedPtr sslCtx() { return nullptr; }
+
+  /*
+   * @return the ClientContextConfig, or absl::nullopt for non-TLS factories.
+   */
+  virtual OptRef<const Ssl::ClientContextConfig> clientContextConfig() const { return {}; }
+
+  /*
+   * @return the QuicCryptoClientConfig or nullptr for non-QUIC factories.
+   */
+  virtual std::shared_ptr<quic::QuicCryptoClientConfig> getCryptoConfig() { return nullptr; }
```

Comment:
How does this work for FIPS build?

Reply:
I think it works because of the forward declaration on line 19 of this file. But I'm a little worried about pulling quiche types into the interface definitions. But it's probably fine, given that it's just a forward declaration and not a real type.

Reply:
yeah, the compile time options build verifies that the !HTTP3 build is fine :-)